### PR TITLE
Feature/start date

### PR DESCRIPTION
**Pull request by sampettersson**

### CODE REVIEW

File: src/new-components/DateInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    </CalendarMonth>
+  ))
+
+  if (calendars.length) {
+    const backProps = getBackProps({ calendars })
+    const forwardProps = getForwardProps({ calendars })
+
+    return (
+      <CalendarContainer
+        style={{ maxWidth: 800, margin: '0 auto', textAlign: 'center' }}
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/new-components/DateInput/index.tsx]: hehe no

File: src/pages/Embark/createQuote.ts [hunk 2]
```diff
@@ -88,5 +88,7 @@ export const createQuote = (storage: any) => async (
     })
   }
 
+  console.log(result.data!.createQuote!.id)
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/pages/Embark/createQuote.ts]: woops 😄

File: src/pages/OfferNew/Introduction/Sidebar/CalendarIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react'
+
+interface CalendarIconProps {
+  color: string
+}
+
+export const CalendarIcon: React.FC<CalendarIconProps> = ({ color }) => (
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/pages/OfferNew/Introduction/Sidebar/CalendarIcon.tsx]: didn't get it to work 😫

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 4]
```diff
@@ -50,27 +72,83 @@ const Value = styled.div`
   }
 `
 
-const TooltipWrapper = styled.div`
-  margin-left: 0.5rem;
+const HandleSwitchingWrapper = styled.div`
+  margin-top: 10px;
+  display: flex;
+  justify-content: space-between;
+  padding: 0 20px;
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: it's now `rem` 😬

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 5]
```diff
@@ -50,27 +72,83 @@ const Value = styled.div`
...
+          <Title>{textKeys.SIDEBAR_STARTDATE_CELL_LABEL()}</Title>
+          <SubTitle>{textKeys.SIDEBAR_STARTDATE_CELL_SUBLABEL()}</SubTitle>
+        </Label>
+        <Value>
+          <DateLabel>
+            {dateValue
+              ? isToday(dateValue)
+                ? textKeys.SIDEBAR_STARTDATE_CELL_VALUE_NEW()
+                : format(dateValue, 'dd MMM yyyy')
+              : textKeys.SIDEBAR_STARTDATE_CELL_VALUE_SWITCHER()}
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: haha yes

File: src/types/dayzed.d.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+declare module 'dayzed' {
```
[the remark this answers is not in the document]
Comment (sampettersson) [reply src/types/dayzed.d.ts]: iIwanted to use the hook at first which wasn't in the types, but I ended up not using the hook, so yes we can use the `@types/dayzed` package 😄

File: src/generated/graphql.tsx [hunk 7]
```diff
@@ -3525,24 +3525,31 @@ export type MemberOfferQuery = { __typename?: 'Query' } & {
   lastQuoteOfMember:
     | ({ __typename?: 'CompleteQuote' } & Pick<
         CompleteQuote,
-        'id' | 'startDate'
```
Comment (palmenhq) [inline src/generated/graphql.tsx]: no `startDate` anymore? :c

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 8]
```diff
@@ -50,27 +77,117 @@ const Value = styled.div`
...
+        setOpen={setDatePickerOpen}
+        date={dateValue || new Date()}
+        setDate={(newDateValue) => {
+          setDateValue(newDateValue)
+          setStartDate({
+            variables: {
+              quoteId: offerId,
+              date: format(newDateValue, gqlDateFormat),
+            },
+          })
```
Comment (palmenhq) [inline src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: just a thought.. should we have some error handling? at least a toast/state/message/modal/even alert.. RN there'll be exactly no feedback to the user if this mutation fails right?

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 9]
```diff
@@ -50,27 +77,117 @@ const Value = styled.div`
...
+                    quoteId: offerId,
+                  },
+                })
+              } else {
+                setStartDate({
+                  variables: {
+                    quoteId: offerId,
+                    date: format(new Date(), gqlDateFormat),
+                  },
+                })
```
Comment (palmenhq) [inline src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: aaand same, error handling?

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 10]
```diff
@@ -50,27 +77,117 @@ const Value = styled.div`
...
+      <RowButton
+        datePickerOpen={datePickerOpen}
+        onClick={() => setDatePickerOpen(!datePickerOpen)}
+      >
+        <Label>
+          <Title>{textKeys.SIDEBAR_STARTDATE_CELL_LABEL()}</Title>
+          <SubTitle>{textKeys.SIDEBAR_STARTDATE_CELL_SUBLABEL()}</SubTitle>
+        </Label>
+        <Value>
+          <DateLabel>{getDateLabel()}</DateLabel>
```
Comment (palmenhq) [inline src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: just an idea, shall we maybe pull this from the quote instead? In that way we could at least show the "actual" date if there'd be errors

**Pull request merged**